src/algorithm/regressor.hpp

```diff
@@ -28,12 +46,99 @@ namespace pinocchio
+    {
+        return ::pinocchio::computeStaticRegressor(model,data,q);
+    }
   }
+
+  ///
+  /// \brief Computes the regressor for the dynamic parameters of a single rigid body.
+  ///
+  /// The result is such that
+  /// \f$ I a + v \times I v = bodyRegressor(v,a) * I.toDynamicParameters() \f$
+  ///
+  /// \param[in] v Velocity of the rigid body
+  /// \param[in] a Acceleration of the rigid body
+  ///
+  /// \return The regressor of the body.
+  ///
+  template<typename MotionVelocity, typename MotionAcceleration>
+  inline Eigen::Matrix<typename MotionVelocity::Scalar,6,10,PINOCCHIO_EIGEN_PLAIN_TYPE(typename MotionVelocity::Vector3)::Options>
+  bodyRegressor(const MotionDense<MotionVelocity> & v, const MotionDense<MotionAcceleration> & a);
+
+  ///
+  /// \brief Computes the regressor for the dynamic parameters of a rigid body attached to a given joint.
+  ///
+  /// This algorithm assumes RNEA has been run to compute the acceleration and gravitational effects.
+  ///
+  /// The result is such that
+  /// \f$ f = jointBodyRegressor(model,data,jointId) * I.toDynamicParameters() \f$
+  /// where \f$ f \f$ is the net force acting on the body, including gravity
+  ///
+  /// \param[in] model The model structure of the rigid body system.
+  /// \param[in] data The data structure of the rigid body system.
+  /// \param[in] jointId The id of the joint.
+  ///
+  /// \return The regressor of the body.
+  ///
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl>
+  inline Eigen::Matrix<Scalar,6,10,Options>
+  jointBodyRegressor(const ModelTpl<Scalar,Options,JointCollectionTpl> & model,
+                     DataTpl<Scalar,Options,JointCollectionTpl> & data,
+                     JointIndex jointId);
+
+  ///
+  /// \brief Computes the regressor for the dynamic parameters of a rigid body attached to a given frame.
+  ///
+  /// This algorithm assumes RNEA has been run to compute the acceleration and gravitational effects.
+  ///
+  /// The result is such that
+  /// \f$ f = frameBodyRegressor(model,data,frameId) * I.toDynamicParameters() \f$
+  /// where \f$ f \f$ is the net force acting on the body, including gravity
+  ///
+  /// \param[in] model The model structure of the rigid body system.
+  /// \param[in] data The data structure of the rigid body system.
+  /// \param[in] frameId The id of the frame.
+  ///
+  /// \return The regressor of the body.
+  ///
+  template<typename Scalar, int Options, template<typename,int> class JointCollectionTpl>
+  inline Eigen::Matrix<Scalar,6,10,Options>
+  frameBodyRegressor(const ModelTpl<Scalar,Options,JointCollectionTpl> & model,
+                     DataTpl<Scalar,Options,JointCollectionTpl> & data,
+                     FrameIndex frameId);
+
+    ///
+    /// \brief Computes the joint torque regressor that links the joint torque
+    ///        to the dynamic parameters of each link according to the current the robot motion.
+    ///
```

Comment:
Can you add a single line to explain the math here. Similarly of what you have done previously in the doc.